apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -410,30 +416,19 @@ func (api *UserManagerAPI) ModelUserInfo(ctx context.Context, args params.Entiti
 
 func (api *UserManagerAPI) modelUserInfo(ctx context.Context, modelTag names.ModelTag) ([]params.ModelUserInfoResult, error) {
 	var results []params.ModelUserInfoResult
-	model, closer, err := api.pool.GetModel(modelTag.Id())
-	if err != nil {
-		return results, errors.Trace(err)
-	}
-	defer closer.Release()
-	if err := api.checkCanRead(ctx, model.ModelTag()); err != nil {
+	if err := api.checkCanRead(ctx, modelTag); err != nil {
 		return results, err
 	}
 
-	users, err := model.Users()
+	modelUserInfo, err := common.ModelUserInfo(ctx, api.accessService, api.apiUserTag, modelTag)
 	if err != nil {
 		return results, errors.Trace(err)
 	}
 
-	for _, user := range users {
-		var result params.ModelUserInfoResult
-		userInfo, err := common.ModelUserInfo(user, model)
-		if err != nil {
-			result.Error = apiservererrors.ServerError(err)
-		} else {
-			userInfo.ModelTag = modelTag.String()
-			result.Result = &userInfo
-		}
-		results = append(results, result)
+	for _, mi := range modelUserInfo {
+		results = append(results, params.ModelUserInfoResult{
+			Result: &mi,
+		})
```

Comment:
This is a behavioural change to the API, what does the `api/client` changes look like i.e. in the prior version the `params.ModelUserInfoResult` return individual errors for each failure. Now it's up front. What does that cost us?

Reply:
So before this change, the only case where an error gets put in there is when one is returned from `common.ModelUserInfo`. This will only returned an error when the access was invalid (which would mean there was invalid access level in the DB, or there was some issue getting the model last login (if there was no model last login no error was returned, that case was handled).

In my opinion the parameter on the facade should really be changed here, to not allow errors for individual users. This is already the case in the `modelmanager` which was also using the same function.

Should I change the facade and rev the version? (I haven't done this before so may need some guidance)

Reply:
As long as we've audited the API client and ensured that we handled the error case correctly, and understood it's a behavioural change, I'm happy to sign this off.

Reply:
[This](https://github.com/juju/juju/blob/d6a9213f1896b1c350028b54fcbd4f33d3c7bd6b/api/client/usermanager/client.go#L175) is the only place in the client we user it. We already don't check for the errors here, just for the results being nil. As far as I can tell this change causes no problems.